packages/grid/x-data-grid-pro/src/internals/hooks/features/export/useGridExcelExport.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import type { Workbook } from 'exceljs';
+import {
+  useGridApiMethod,
+  useGridLogger,
+  unstable_useGridRegisterPreProcessor as useGridRegisterPreProcessor,
+} from '@mui/x-data-grid';
+// import { buildWarning } from '@mui/x-data-grid/internals'
+// import { exportAs } from '@mui/x-data-grid/internals'
+import { DataGridProProps } from '../../../models/dataGridProProps';
+import { GridApiPro } from '../../../models/gridApiPro';
+import {
+  GridExcelExportApi,
+  GridExcelExportOptions,
+  GridExportExtension,
+  GridToolbarExportProps,
+} from './gridExcelExportInterface';
+import { buildExcel } from './serializer/excelSerializer';
+import { defaultGetRowsToExport, getColumns, buildWarning } from './serializer/utils';
+import { GridExcelExportMenuItem } from '../../../components';
+
+// TODO: import the one from community when bundling strategy is in its finale version
+function exportAs<ExtraExtensions extends string>(
+  blob: Blob,
+  extension: GridExportExtension | ExtraExtensions = 'csv',
+  filename: string = document.title,
+): void {
+  const fullName = `${filename}.${extension}`;
+
+  // Test download attribute first
+  // https://github.com/eligrey/FileSaver.js/issues/193
+  if ('download' in HTMLAnchorElement.prototype) {
+    // Create an object URL for the blob object
+    const url = URL.createObjectURL(blob);
+
+    // Create a new anchor element
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = fullName;
+
+    // Programmatically trigger a click on the anchor element
+    // Useful if you want the download to happen automatically
+    // Without attaching the anchor element to the DOM
+    a.click();
+
+    // https://github.com/eligrey/FileSaver.js/issues/205
+    setTimeout(() => {
+      URL.revokeObjectURL(url);
+    });
+    return;
+  }
+
+  throw new Error('MUI: exportAs not supported');
+}
+
+/**
+ * @requires useGridColumns (state)
+ * @requires useGridFilter (state)
+ * @requires useGridSorting (state)
+ * @requires useGridSelection (state)
+ * @requires useGridParamsApi (method)
+ */
+export const useGridExcelExport = (
+  apiRef: React.MutableRefObject<GridApiPro>,
+  props: Partial<Pick<DataGridProProps, 'experimentalFeatures'>>,
+): void => {
+  const featureIsActivated = props?.experimentalFeatures?.excelExport === true;
+
+  const logger = useGridLogger(apiRef, 'useGridExcelExport');
+  const inactivatedError = buildWarning(
+    [
+      'MUI: export .excel is experimental and must be activated by setting',
+      '<DataGridPro experimentalFeatures={{excelExport: true}} />',
+    ],
+    'error',
+  );
+
+  const getDataAsExcel = React.useCallback(
```

Comment:
```suggestion
  const getDataAsExcel = React.useCallback<GridExcelExportApi['getDataAsExcel']>(
```

Same for `exportDataAsExcel`
In the past we had slight interface differences between the `GridApi` signature and the column definition due to those type explicit writting.